collect_app/src/main/java/org/odk/collect/android/dao/helpers/ContentResolverHelper.java

```diff
@@ -76,20 +79,43 @@ public static String getFormPath(Uri uri) {
         return formPath;
     }
 
-    /**
-     * Using contentResolver to get a file's extension by the uri
-     *
-     * @param fileUri Whose name we want to get
-     * @return The file's extension without a dot eg. "mp3" not ".mp3"
-     */
-    public static String getFileExtensionFromUri(Context context, Uri fileUri) {
-        String fileName = new MediaUtils().getFileNameFromUri(context, fileUri);
-        if (fileName != null && fileName.contains(".")) {
-            return fileName.substring(fileName.lastIndexOf('.') + 1);
-        } else {
-            return fileUri.getScheme() != null && fileUri.getScheme().equals(ContentResolver.SCHEME_CONTENT)
-                    ? MimeTypeMap.getSingleton().getExtensionFromMimeType(context.getContentResolver().getType(fileUri))
-                    : MimeTypeMap.getFileExtensionFromUrl(fileUri.toString());
+    public static String getFileExtensionFromUri(Uri fileUri) {
+        String mimeType = getContentResolver().getType(fileUri);
+
+        String extension = fileUri.getScheme() != null && fileUri.getScheme().equals(ContentResolver.SCHEME_CONTENT)
```

Comment:
Or was that the selfie video case again?

Reply:
This was the fix added for a crash some time ago https://github.com/getodk/collect/commit/f98b98dddc3cc9af8e1db5144f4c9c7313854df1 so I wanted to keep it just in case seeing how other cases might be tricky.

Reply:
Got it. Would be great to follow up with #2834 because this method has gone through lots of big changes during this PR that probably would have been avoided with tests.